ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -224,363 +301,231 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
-      ) extends LoggingTransactionErrorImpl(
-            cause = LookupError.MissingPackage.pretty(packageId, reference)
-          )
-
+      ) extends LoggingTransactionErrorImpl(cause = message)
     }
 
     @Explanation(
-      """This error occurs if a package referred to by a Command fails validation. This should not happen as packages are validated when being uploaded."""
+      """Every ledger Api command contains a ledger-id which is verifying against the running ledger.
+          This error indicates that the provided ledger-id does not match the expected one."""
     )
-    @Resolution("Contact support.")
-    object PackageValidationFailed
-        extends ErrorCode(
-          id = "PACKAGE_VALIDATION_FAILED",
-          ErrorCategory.MaliciousOrFaultyBehaviour,
-        ) {
-      case class Reject(validationErrorCause: String)(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = validationErrorCause
-          )
-    }
-  }
-
-  object PreprocessingErrors extends ErrorGroup {
-    @Explanation("""This error occurs if a command fails during interpreter pre-processing.""")
-    @Resolution("Inspect error details and correct your application.")
-    object PreprocessingFailed
-        extends ErrorCode(
-          id = "COMMAND_PREPROCESSING_FAILED",
-          ErrorCategory.InvalidIndependentOfSystemState,
-        ) {
-      case class Reject(
-          err: LfError.Preprocessing.Error
-      )(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = err.message
-          )
-    }
-  }
-
-  object InterpreterErrors extends ErrorGroup {
-
-    @Explanation("""This error occurs if the Daml transaction failed during interpretation.""")
-    @Resolution("This error type occurs if there is an application error.")
-    object GenericInterpretationError
+    @Resolution("Ensure that your application is correctly configured to use the correct ledger.")
+    object LedgerIdMismatch
         extends ErrorCode(
-          id = "DAML_INTERPRETATION_ERROR",
-          ErrorCategory.InvalidGivenCurrentSystemStateOther,
+          id = "LEDGER_ID_MISMATCH",
+          ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
         ) {
-
-      case class Error(override val cause: String)(implicit
+      case class Reject(override val cause: String)(implicit
           loggingContext: ContextualizedErrorLogger
       ) extends LoggingTransactionErrorImpl(
-            cause = cause
+            cause = cause,
+            definiteAnswer = true,
           )
-
     }
 
     @Explanation(
-      """This error occurs if the Daml transaction failed during interpretation due to an invalid argument."""
+      """This error is emitted, when a submitted ledger Api command could not be successfully deserialized due to mandatory fields not being set."""
```

Comment:
```suggestion
      """This error is emitted when a submitted ledger Api command has some mandatory fields not set."""
```
as I'm not sure if mentioning `deserialized` will be useful to the users (as it seems like an implementation detail)